webapp/components/utils/PostHelpers.js

```diff
@@ -0,0 +1,35 @@
+import PostMutations from '~/graphql/PostMutations.js'
+
+export function postMenuModalsData(truncatedPostName, confirmCallback, cancelCallback = () => {}) {
+  return {
+    delete: {
+      titleIdent: 'delete.contribution.title',
+      messageIdent: 'delete.contribution.message',
+      messageParams: {
+        name: truncatedPostName,
+      },
+      buttons: {
+        confirm: {
+          danger: true,
+          icon: 'trash',
+          textIdent: 'delete.submit',
+          callback: confirmCallback,
+        },
+        cancel: {
+          icon: 'close',
+          textIdent: 'delete.cancel',
+          callback: cancelCallback,
+        },
+      },
+    },
+  }
+}
+
+export function deletePostMutation(postId) {
+  return {
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-06-14T10:56:17Z" title="Friday, June 14th 2019, 12:56:17 pm +02:00">Jun 14, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,43 @@
+import gql from 'graphql-tag'
+
+export function postMenuModalsData(truncatedPostName, confirmCallback, cancelCallback = () => {}) {
+  return {
+    delete: {
+      titleIdent: 'delete.contribution.title',
+      messageIdent: 'delete.contribution.message',
+      messageParams: {
+        name: truncatedPostName,
+      },
+      buttons: {
+        confirm: {
+          danger: true,
+          icon: 'trash',
+          textIdent: 'delete.submit',
+          callback: confirmCallback,
+        },
+        cancel: {
+          icon: 'close',
+          textIdent: 'delete.cancel',
+          callback: cancelCallback,
+        },
+      },
+    },
+  }
+}
+
+export function deletePostMutation(postId) {
+  // TODO: Replace "gqlMutation" with "DeletePost" from '~/graphql/PostMutations.js', has not worked for me.
```


It's probably not a big deal, but if you want to have a go at refactoring it together later, we can try again :smile:

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-06-14T11:25:36Z" title="Friday, June 14th 2019, 1:25:36 pm +02:00">Jun 14, 2019</time>_

---


I would love to try @mattwr18 .

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-06-14T16:41:35Z" title="Friday, June 14th 2019, 6:41:35 pm +02:00">Jun 14, 2019</time>_

---


We have done. 💫 @mattwr18